packages/automerge-repo-network-websocket/src/NodeWSServerAdapter.ts

```diff
@@ -116,9 +119,27 @@ export class NodeWSServerAdapter extends NetworkAdapter {
 
         // In this client-server connection, there's only ever one peer: us!
         // (and we pretend to be joined to every channel)
-        socket.send(
-          CBOR.encode({ type: "peer", senderId: this.peerId, channelId })
-        )
+        if (supportedProtocolVersions === undefined) {
+          socket.send(CBOR.encode({ type: "error", errorMessage: "missing supportedProtocolVersions"}))
+          this.sockets[senderId].close()
```

Comment:
The client adapter is set up to try to reconnect automatically whenever the socket is closed. If we close the socket from the server end, this results in an infinite loop of attempting to rejoin, rejection because of the version and the socket being closed again. 

Might it be better if the socket is closed on the client end once it realises it cannot connect?